airbyte-integrations/connectors/source-cockroachdb/src/main/java/io/airbyte/integrations/source/cockroachdb/CockroachDbSource.java

```diff
@@ -107,22 +103,18 @@ protected boolean isNotInternalSchema(final JsonNode jsonNode, final Set<String>
   }
 
   @Override
-  protected DataSource createDataSource(final JsonNode sourceConfig) {
+  public JdbcDatabase createDatabase(final JsonNode sourceConfig) throws SQLException {
     final JsonNode jdbcConfig = toDatabaseConfig(sourceConfig);
 
+    // Create the JDBC data source
     final DataSource dataSource = DataSourceFactory.create(
         jdbcConfig.get(JdbcUtils.USERNAME_KEY).asText(),
         jdbcConfig.has(JdbcUtils.PASSWORD_KEY) ? jdbcConfig.get(JdbcUtils.PASSWORD_KEY).asText() : null,
         driverClass,
         jdbcConfig.get(JdbcUtils.JDBC_URL_KEY).asText(),
         JdbcUtils.parseJdbcParameters(jdbcConfig, JdbcUtils.CONNECTION_PROPERTIES_KEY));
     dataSources.add(dataSource);
-    return dataSource;
-  }
 
-  @Override
-  public JdbcDatabase createDatabase(final JsonNode sourceConfig) throws SQLException {
-    final DataSource dataSource = createDataSource(sourceConfig);
     final JdbcDatabase database = new DefaultJdbcDatabase(dataSource, sourceOperations);
     quoteString = (quoteString == null ? database.getMetaData().getIdentifierQuoteString() : quoteString);
```

Comment:
Weirdly - no since it's an instance variable. Ideally, we'll want to determine whether this should be a quote string or not but for now will keep it to reduce the scope of this change